### ISSUE TITLE
Setting accepting NEW contracts to false prevents SIA from using EXISTING contract

### ISSUE THREAD

**Issue by MaxMaximus** [NebulousLabs/Sia#2415]
Hello

I have found out disabling of accepting **new** file contract on host prevent SIA from using already **existing** file contracts.
I set up one host this summer for testing. I got 15 file contracts and was getting some real upload and download traffic.
After i finished tests I set "acceptingcontracts = false" to finish work of host correctly. But after this setting all activity on existing contracts (except of storage proofs) immediately stopped. Host is still running about 1.5 month in 24/7 online mode but did not get even single piece of upload/download data. Only a lot of settings update (which shows that host is running and reachable by other clients).

Here my stat from host (siad ver 1.3.0)
Host Financials:
        Contract Count:               15
        Transaction Fee Compensation: 0 H
        Potential Fee Compensation:   15 SC
        Transaction Fee Expenses:     0 H

        Storage Revenue:           0 H
        Potential Storage Revenue: 1.029 SC

        Locked Collateral: 349.2 SC
        Risked Collateral: 2.059 SC
        Lost Collateral:   0 H

        Download Revenue:           0 H
        Potential Download Revenue: 945.6 uS
        Upload Revenue:             0 H
        Potential Upload Revenue:   3.464 mS

RPC Stats:
        Error Calls:        0
        Unrecognized Calls: 2479
        Download Calls:     0
        Renew Calls:        0
        Revise Calls:       0
        Settings Calls:     67992
        FormContract Calls: 0
        
I am not sure is this issue on host on renter side. But it is serious. 
This setting should not affect work of existing contracts. They are still valid and can be used by renter until expire or at least until renew time. It should affect only forming new contract and renew of contracts.
In current state it is not only cause additional expenses for renters (they need to form a new contract with other host while already have valid and thus pay additional fees and lock up additional allowance) and lost revenues to host, but also open serious attack vector on network in general.  
I do not wite it here or it will help exploit it(but i think some users already exploiting it on minor scale). 
I can send  it in private to devs If you do not understand which exactly vulnerability it opens.

P.S.
I think similar issue happens when host owner rise prices in host setting after successful formation of some contacts. Host tries to apply this new prices to old contracts (which signed when prices were lower) and renters refuse to accept this change. After each price change I saw a lot of errors in logs where  my host and renters "argue" about price and revise attempt (renters try to upload/download) failed due to new prices.  
This also affectively "kills" all existing contracts after host setting change.
While logical solution will be apply new prices only to new/renewed contracts. And keep old condition for old contracts until they expire. 
It is a contract (signed agreement about storage condition including prices) between two sides after all, isn't it?

**Comment by lukechampine** [comment]
I don't see evidence in the host source code that suggests that `AcceptingContracts` affects anything other than contract formation. So if there is a problem, it's on the renter side. It's possible that the renter is disregarding previously-formed contracts if the host is now marked as not accepting contracts; however, I wasn't able to find anything in the renter source code that supports that hypothesis. The `AcceptingContracts` setting only affects which hosts are reported by the hostdb; it doesn't affect which contracts the renter uses for uploads.
However, the renter code has changed a lot in recent months. It's possible that there was a bug in an earlier version that caused the renter to not use such hosts. The easiest way to confirm this would be to write a test that recreates the scenario you described. Then we can run the test on multiple versions of the code to see if the bug appears.

For vulnerability disclosures, you can message me (nemo) on our discord server.

**Comment by MaxMaximus** [comment]
Yes, I thought this over again - first issue very likely is completely on the renter side.
I have checked logs of my host and did not find any errors after date when i set up AcceptingContracts to false.
Zero RPC stats counter also suggest this
Revise Calls: 0 + Download Calls: 0 + Error Calls: 0 (collected after ~1 month of uptime)
There is no any calls from owners of existing contracts - only routine settings calls (scans). 
While AcceptingContract was set to true there were few dozen or hundreds of revise(mostly uploads) calls almost each day.

So all looks like renters who own contracts with host simply stop making calls/drop host from active contracts list if see AcceptingContract=false flag.  Not sure about version though. Or all renters with whom i have contracts still use old clients, or last official siad release (1.3.0) still have this issue in renter module.  

Second issue on the other hand very likely  on host side, because after each setting change which include increase of host pricing or lowering of collateral i saw a lot of errors in logs like this:

> 2017/07/12 18:30:32.485613 errors.go:151: communication error: error with x.x.x.x:61971: incoming RPCReviseContract failed: revision iteration failed: rejected proposed modifications: unable to verify updated contract: expected at least 1310914180732619325440 to be exchanged, but 1275711775622084566352 was exchanged: rejected for high paying renter valid output

> 2017/07/12 18:38:38.842244 errors.go:151: communication error: error with x.x.x.x:2256: incoming RPCReviseContract failed: revision iteration failed: rejected proposed modifications: unable to verify updated contract: expected at least 1310817090362249576448 to be exchanged, but 1275614491070974077861 was exchanged: rejected for high paying renter valid output

> 2017/07/14 01:28:53.504762 errors.go:151: communication error: error with x.x.x.x:62661: incoming RPCReviseContract failed: revision iteration failed: rejected proposed modifications: unable to verify updated contract: host expected to post at most 1561407336286303289344 collateral, but contract has host posting 1572043780541050028117: rejected for low paying host missed output

> 2017/07/18 06:32:05.361709 errors.go:151: communication error: error with x.x.x.x:55813: incoming RPCReviseContract failed: revision iteration failed: rejected proposed modifications: unable to verify updated contract: host expected to post at most 2530369232576398229504 collateral, but contract has host posting 2535579490211920434897: rejected for low paying host missed output

> 2017/07/20 08:07:28.252178 errors.go:151: communication error: error with x.x.x.x:52078: incoming RPCReviseContract failed: revision iteration failed: rejected proposed modifications: unable to verify updated contract: host expected to post at most 2476386986650817789952 collateral, but contract has host posting 2514456120872796365239: rejected for low paying host missed output

> 2017/07/20 18:00:37.539087 errors.go:151: communication error: error with x.x.x.x:52497: incoming RPCReviseContract failed: revision iteration failed: rejected proposed modifications: unable to verify updated contract: host expected to post at most 2390753279984699179008 collateral, but contract has host posting 2490232073465543991778: rejected for low paying host missed output

I did not check this behavior on latest release, but previous (1.2.2) seems tried to force new price/collateral on all contracts both new/revised and old. Is this already fixed? Or currently it is a expected behavior for hosts for some reason?

P.S.
I will send vulnerability detail on Discord later.

**Comment by avahowell** [comment]
@MaxMaximus can you post your host settings? You may have been affected by #2450. #2449 proves that accepting settings `acceptingcontracts = false` does not prevent the host from serving downloads or submitting storage proofs.

**Comment by MaxMaximus** [comment]
I have posed it already in another issues, so i just put link here: https://github.com/NebulousLabs/Sia/files/1412656/host-v.txt
There is a siac host -v command output including host settings grabbed at different times. Look like all OK, and no signs of bug you mentioned. 

Settings in host.json file looks OK to:

> 	"settings": {
		"acceptingcontracts": false,
		"maxdownloadbatchsize": 17825792,
		"maxduration": 16128,
		"maxrevisebatchsize": 17825792,
		"netaddress": "domain.name:9982",
		"windowsize": 144,
		"collateral": "46296296296",
		"collateralbudget": "10000000000000000000000000000",
		"maxcollateral": "20000000000000000000000000",
		"mincontractprice": "1000000000000",
		"mindownloadbandwidthprice": "25000000000000",
		"minstorageprice": "11574074074",
		"minuploadbandwidthprice": "1000000000000"
	},

I do not remember initial setting i have used. But it was similar.
Obviously acceptingcontracts = true initially (set to false after 28.07.2017)
And i changed maxcollateral per contract from default value to 20 SC Per Contract later.
Also I changed prices few times in July but did not touch them after.

As i already wrote in previous post - it does not looks like an error on host side but more on renter side. There is no any errors in logs or in RPC stats. But for some reason all renter started to avoid my host immediately after I set acceptingcontracts = false. Just no any single incoming request for uploads/downloads after setting change.

And yes - this does not affect storage proofs. Host posted at least few correct storage proofs after setting apply. So only affect upload/download(revise) calls on existing contracts.
